src/Stratis.Bitcoin.Features.Consensus/PowConsensusValidator.cs

```diff
@@ -284,426 +599,300 @@ public virtual void ExecuteBlock(ContextInformation context, TaskScheduler taskS
+            for (int i = 0; i < lenght; i++)
             {
                 if (a[i] != b[i])
                     return false;
             }
             return true;
         }
 
-        private uint256 BlockWitnessMerkleRoot(Block block, ref bool mutated)
+        /// <summary>
+        /// Calculates merkle root for witness data.
+        /// </summary>
+        /// <param name="block">Block which transactions witness data is used for calculation.</param>
+        /// <param name="mutated"><c>true</c> if at least one leaf of the merkle tree has the same hash as any subtree. Otherwise: <c>false</c>.</param>
+        /// <returns>Merkle root.</returns>
+        private uint256 blockWitnessMerkleRoot(Block block, out bool mutated)
         {
             List<uint256> leaves = new List<uint256>();
             leaves.Add(uint256.Zero); // The witness hash of the coinbase is 0.
             foreach (Transaction tx in block.Transactions.Skip(1))
                 leaves.Add(tx.GetWitHash());
 
-            return this.ComputeMerkleRoot(leaves, ref mutated);
+            return this.computeMerkleRoot(leaves, out mutated);
         }
 
-        private uint256 BlockMerkleRoot(Block block, ref bool mutated)
+        /// <summary>
+        /// Calculates merkle root for block's trasnactions.
+        /// </summary>
+        /// <param name="block">Block which transactions are used for calculation.</param>
+        /// <param name="mutated"><c>true</c> if block contains repeating sequences of transactions without affecting the merkle root of a block. Otherwise: <c>false</c>.</param>
+        /// <returns>Merkle root.</returns>
+        private uint256 blockMerkleRoot(Block block, out bool mutated)
         {
             List<uint256> leaves = new List<uint256>(block.Transactions.Count);
             for (int s = 0; s < block.Transactions.Count; s++)
                 leaves.Add(block.Transactions[s].GetHash());
 
-            return this.ComputeMerkleRoot(leaves, ref mutated);
-        }
-
-        private uint256 ComputeMerkleRoot(List<uint256> leaves, ref bool mutated)
-        {
-            uint256 hash = null;
-            this.MerkleComputation(leaves, ref hash, ref mutated, -1, null);
-            return hash;
+            return this.computeMerkleRoot(leaves, out mutated);
         }
 
-        private void MerkleComputation(List<uint256> leaves, ref uint256 root, ref bool pmutated, int branchpos, List<uint256> pbranch)
+        /// <summary>
+        /// Computes merkle root.
+        /// </summary>
+        /// <remarks>This implements a constant-space merkle root/path calculator, limited to 2^32 leaves.</remarks>
+        /// <param name="leaves">Merkle tree leaves.</param>
+        /// <param name="mutated"><c>true</c> if at least one leaf of the merkle tree has the same hash as any subtree. Otherwise: <c>false</c>.</param>
+        private uint256 computeMerkleRoot(List<uint256> leaves, out bool mutated)
```

Comment:
methods are always with uppercase first letter regardless of whether they are private/public/protected...